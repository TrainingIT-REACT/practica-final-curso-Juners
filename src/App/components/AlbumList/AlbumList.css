.album-list .album a {
  padding: 5px 15px;
}

.album-list .album:hover {
  background-color: lightgrey;
}