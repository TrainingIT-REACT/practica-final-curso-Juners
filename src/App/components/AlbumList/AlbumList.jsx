import React from "react";
import { Link } from "react-router-dom";
import "./AlbumList.css";

const Album = ({ album = {} }) => {
  return (
    <div className="album grid-container full">
      <Link to={`/album/${album.id}`} className="grid-x">
        {album.name}
      </Link>
    </div>
  );
};

const AlbumList = ({ albums = [] }) => {
  return (
    <div className="album-list">
      {albums.map(album => (
        <Album key={album.id} album={album} />
      ))}
    </div>
  );
};

export default AlbumList;
