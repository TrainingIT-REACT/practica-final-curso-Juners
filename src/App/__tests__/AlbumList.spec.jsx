import React from 'react';
import { render } from 'enzyme';

import AlbumList from '../components/AlbumList';
import { MemoryRouter } from 'react-router-dom';

describe(AlbumList, () => {
  describe("Render", () => {

    it('renders correctly', () => {
      const wrapper = render(
        <MemoryRouter>
          <AlbumList albums={[
            { id: 1, name: "first" },
            { id: 2, name: "second" }
          ]} />
        </MemoryRouter>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it("has two elements", () => {
      const wrapper = render(
        <MemoryRouter>
          <AlbumList albums={[
            { id: 1, name: "first" },
            { id: 2, name: "second" }
          ]} />
        </MemoryRouter>
      );
      expect(wrapper.find("div.album").length).toBe(2);
      expect(wrapper.find("div.album").first().text()).toBe("first");
      expect(wrapper.find("div.album").last().text()).toBe("second");
    });

    it("has no elements", () => {
      const wrapper = render(
        <MemoryRouter>
          <AlbumList />
        </MemoryRouter>
      );
      expect(wrapper.find("div.album").length).toBe(0);
    });

  });
});