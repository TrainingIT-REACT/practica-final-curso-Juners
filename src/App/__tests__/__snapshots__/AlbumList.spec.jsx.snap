// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumList Render renders correctly 1`] = `
<div
  class="album-list"
>
  <div
    class="album grid-container full"
  >
    <a
      class="grid-x"
      href="/album/1"
    >
      first
    </a>
  </div>
  <div
    class="album grid-container full"
  >
    <a
      class="grid-x"
      href="/album/2"
    >
      second
    </a>
  </div>
</div>
`;
